symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -54,6 +137,58 @@ public V4Message send(String streamId, String message) {
     }
   }
 
+  /**
+   * Retrieved the details of a message given its message ID.
+   *
+   * @param messageId the ID of the message to be retrieved
+   * @return a {@link V4Message} containing the message's details, null if the message was not found
+   * @see <a href="https://developers.symphony.com/restapi/reference#get-message-v1">Get Message v1</a>
+   */
+  public V4Message getMessage(@Nonnull String messageId) {
+    try {
+      return this.messagesApi.v1MessageIdGet(authSession.getSessionToken(), authSession.getKeyManagerToken(),
+          messageId);
+    } catch (ApiException e) {
+      log.error("Cannot get message", e);
+      sleep(1_000);
+      try {
+        authSession.refresh();
+      } catch (AuthUnauthorizedException exception) {
+        log.error("Cannot authenticate", exception);
+        return null;
+      }
+      return this.getMessage(messageId);
+    }
+  }
+
+  /**
+   * Imports a list of messages to Symphony
+   *
+   * @param messages the list of messages to be imported
+   * @return the list of imported messages
+   * @see <a href="https://developers.symphony.com/restapi/reference#import-message-v4">Import Message</a>
+   */
+  public List<V4ImportResponse> importMessages(List<V4ImportedMessage> messages) {
```

Comment:
just curious what's the use case for this (it seems pretty advanced) (@symphony-thibault ?)

Reply:
The doc https://developers.symphony.com/restapi/reference#import-message-v4 :)

Reply:
Yep I read it but could not figure out who would use that (or if it should be exposed in the BDK's service, maybe in the API yes for specific use cases)